test/unit/retabled/internal_test.cljc

```diff
@@ -0,0 +1,128 @@
+(ns retabled.internal-test
+  (:require [clojure.test :refer [deftest is testing]]
+            [retabled.internal :as internal]))
+
+(deftest update-default-entries-per-page!
+  (let [entries-test (fn [expected controls]
+                       (internal/update-default-entries-per-page! controls)
+                       (let [actual (:entries-per-page @internal/PAGING-INFO)]
+                         (is (= expected actual))))]
+    (testing "Correctly updates when paging is nil"
+      (entries-test (:entries-per-page @internal/PAGING-INFO) {}))
+    (testing "Correctly updates when entries-option is nil"
+      (entries-test (:entries-per-page @internal/PAGING-INFO) {:paging true}))
+    (testing "Correctly updates when default and values are nil"
+      (entries-test (:entries-per-page @internal/PAGING-INFO) {:paging {:entries-option true}}))
+    (testing "Correctly updates when default is set and values are nil"
+      (entries-test 20 {:paging {:entries-option {:default 20}}}))
+    (testing "Correctly updates when default is nil and values are set"
+      (entries-test 3 {:paging {:entries-option {:values [3 20 100]}}}))
+    (testing "Correctly updates when default is set and values are set"
+      (entries-test 20 {:paging {:entries-option {:default 20 :values [3 20 100]}}}))
+    #_(testing "Correctly updates when default is nil and first value is nil" ;; not quite built to handle this yet
+      (entries-test (:entries-per-page @internal/PAGING-INFO) {:paging {:entries-option {:values [nil]}}}))))
+
+(deftest update-current-data!
+  (let [current-data-test (fn [expected table-id entries]
+                            (internal/update-current-data! table-id entries)
+                            (let [actual @internal/CURRENT-DATA]
+                              (is (= expected actual))))]
+    #_(testing "Correctly updates when table-id or entries are nil" ;; Should handle nils
+      (current-data-test @internal/CURRENT-DATA nil nil)
+      (current-data-test @internal/CURRENT-DATA "1" nil)
+      (current-data-test @internal/CURRENT-DATA nil [{:first "entry"}{:second "entry"}]))
+    #_(testing "Correctly updates when table-id and entries are set"
+      (current-data-test {"1" [{:first "entry"}{:second "entry"}]} "1" [{:first "entry"}{:second "entry"}]))))
+
+(deftest update-all-data!
+  (let [all-data-test (fn [expected table-id entries]
```

Comment:
add docstrings to the deftest, and also optionally to the is-using test-helper functions